model-optimizer/extensions/middle/ONNXResize11ToInterpolateV4.py

```diff
@@ -0,0 +1,166 @@
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.graph.graph import Graph, Node, rename_nodes
+from mo.ops.const import Const
+from mo.ops.shape import Shape
+from mo.ops.strided_slice import StridedSlice
+
+
+def convert_mode(onnx_mode: str) -> str:
+    return {'nearest': 'nearest', 'linear': 'linear_onnx', 'cubic': 'cubic'}[onnx_mode]
+
+
+def replace_resize(graph: Graph, resize: Node):
+    log.debug("Converting of ONNX Resize-11 to Interpolate-4 is triggered.")
+
+    input_shape = resize.in_port(0).data.get_shape()
+    input_rank = len(input_shape)
+    if input_rank not in {4, 5}:
+        log.warning('The input shape is not 4D or 5D for op with name {}'.format(resize.name))
+        return
+
+    num_of_inputs = len(resize.in_ports())
+    resize_name = resize.name
+    assert num_of_inputs in {3, 4}, \
+        "Number of inputs of ONNXResize (with name {}) should be equal to 3 or 4".format(resize_name)
+
+    assert resize.soft_get('coordinate_transformation_mode') != 'tf_crop_and_resize', \
+        'Mode tf_crop_and_resize is not supported for op {} with name {}'.format(resize.op, resize.name)
+
+    layout = graph.graph['layout']
+
+    if input_rank == 4:
+        begin_dim = get_height_dim(layout, input_rank)
+        end_dim = get_width_dim(layout, input_rank) + 1
+    else:
+        begin_dim = get_depth_dim(layout, input_rank)
+        end_dim = get_width_dim(layout, input_rank) + 1
+
+    sizes_ss = create_op_with_const_inputs(graph, StridedSlice,
+                                           {1: int64_array([begin_dim]),
+                                            2: int64_array([end_dim]),
+                                            3: int64_array([1])},
+                                           {'name': resize_name + '/StridedSlice_',
+                                            'begin_mask': int64_array([1]),
+                                            'end_mask': int64_array([1]),
+                                            'new_axis_mask': int64_array([0]),
+                                            'shrink_axis_mask': int64_array([0]),
+                                            'ellipsis_mask': int64_array([0])})
+    scales_ss = create_op_with_const_inputs(graph, StridedSlice,
+                                            {1: int64_array([begin_dim]),
+                                             2: int64_array([end_dim]),
+                                             3: int64_array([1])},
+                                            {'name': resize_name + '/StridedSlice_',
+                                             'begin_mask': int64_array([1]),
+                                             'end_mask': int64_array([1]),
+                                             'new_axis_mask': int64_array([0]),
+                                             'shrink_axis_mask': int64_array([0]),
+                                             'ellipsis_mask': int64_array([0])})
```

Comment:
This transformation is not reshape-able. I think that @jane-intel will be unhappy.

Moreover, I think that there should be a way to move it to the front phase.

Reply:
But 

> This transformation is not reshape-able. I think that @jane-intel will be unhappy.

But if we will not skip Interpolate-4 in the transformation `InterpolateReshapeWA`, the transformation `ONNXResize11ToInterpolate` is OK.

Reply:
> Moreover, I think that there should be a way to move it to the front phase.
We cannot move this transformation to the front phase, because in this transformation we need to know input rank, and to know input rank, we need to know input shape. But input shapes are unknown in the front phase.